models: add lockdown default configuration files

arnaldo2792 opened the pull request:
**Issue number:**
#813 


**Description of changes:**

```
fc416bf1  docs: update security guidance document
```

This commit updates the security guidance document since the default kernel lockdown mode for newer variants is `integrity`

```
4e6b5a5f models: add lockdown default configuration files
```

This commit adds default configuration files for the different values that can be used for the kernel lockdown setting. It also changes the lockdown mode for the ECS variant from `none` to `integrity`.


**Testing done:**
In aws-dev, aws-ecs, k8s 1.16, 1.17, 1.18, 1.19 x86_64:

- Launch nginx container/task/pod
- Check `systemctl status`, no failing units
- Check the expected values of `/sys/kernel/security/lockdown` in each variant:

#### AWS dev:

```shell
cat /sys/kernel/security/lockdown
[none] integrity confidentiality
```

#### AWS ECS:

```shell
cat /sys/kernel/security/lockdown
none [integrity] confidentiality
```

#### AWS k8s 1.19:

```shell
cat /sys/kernel/security/lockdown
[none] integrity confidentiality
```

#### AWS k8s 1.18:

```shell
cat /sys/kernel/security/lockdown
[none] integrity confidentiality
```

#### AWS k8s 1.17:

```shell
cat /sys/kernel/security/lockdown
[none] integrity confidentiality
```

#### AWS k8s 1.16:

```shell
cat /sys/kernel/security/lockdown
[none] integrity confidentiality
```

#### VMWare dev:
![Screenshot from 2021-04-28 10-58-39](https://user-images.githubusercontent.com/6305870/116451091-e92e4980-a810-11eb-94f3-7998193eb19b.png)


**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

arnaldo2792 committed:
models: add lockdown default configuration files

This commit adds default configuration files for the different values
that can be used for the kernel lockdown setting.

bcressey commented on sources/models/shared-defaults/lockdown-confidentiality.toml:
```diff
@@ -0,0 +1,3 @@
+# Kernel
+[settings.kernel]
+lockdown = "confidentiality"
```
I don't think we want this one - "confidentiality" isn't a setting I'd recommend.

arnaldo2792 commented:
- Removed confidentiality default setting

arnaldo2792 commented:
- Fixed security guidance document

arnaldo2792 committed:
docs: update security guidance document

This commit updates the security guidance document since the default
kernel lockdown mode for newer variants is `integrity`

arnaldo2792 commented:
- Separated changes in two commits
